VEuPathDB/web-eda
# New multiple-variable entity constraints

Resolves https://github.com/VEuPathDB/web-eda/issues/1410

There are two ways to test this PR

1. use your regular back end and this branch as-is. You will only be testing the front-end logic with spoofed back-end responses.
2. deploy a local back end using `DATA_TAG=branch-variable-constraints` and `RSERVE_TAG=3.2.1` and [use this branch from PR #1418 instead](https://github.com/VEuPathDB/web-eda/pull/1418) (which is a complete front and back end solution)

I'd appreciate it if you could cast your eyes over #1418 too.  I will merge that into this branch before merging this one.


Some info about the response spoofing, if you're using that:

There is a [function](https://github.com/VEuPathDB/web-eda/pull/1415/files#diff-b04aa83b90284c989ea432dda24f7697b0c7fe672c974242eb96a018b1d9127a) `temporaryHack()` which reconfigures the regular flat array from the back end into the `string[][]` shape we need, using various hard-coded hacks.  Before reporting any dependency-related issues, please check that the output of this function (logged in the console) is what was expected compared with changes made to the back end [here](https://github.com/VEuPathDB/EdaDataService/pull/216).
